ostd/src/bus/mmio/mod.rs

```diff
@@ -32,7 +28,39 @@ pub static MMIO_BUS: SpinLock<MmioBus> = SpinLock::new(MmioBus::new());
 static IRQS: SpinLock<Vec<IrqLine>> = SpinLock::new(Vec::new());
 
 pub(crate) fn init() {
-    #[cfg(all(target_arch = "x86_64", feature = "cvm_guest"))]
+    #[cfg(target_arch = "riscv64")]
+    riscv64_init();
+
+    #[cfg(target_arch = "x86_64")]
+    x86_microvm_init();
+}
+
+// The following functions need to be refactored because we don't actually want this code in the ostd.
+
+#[cfg(target_arch = "riscv64")]
+fn riscv64_init() {
+    use crate::arch::{boot::DEVICE_TREE, device::plic::enable_external_interrupt};
+
+    let mut mmio_bus = MMIO_BUS.lock();
+    for virtio in DEVICE_TREE
+        .get()
+        .unwrap()
+        .find_all_nodes("/soc/virtio_mmio")
+    {
+        let irq = virtio.interrupts().unwrap().next().unwrap();
+        let region = virtio.reg().unwrap().next().unwrap();
+        let device = MmioCommonDevice::new(
+            region.starting_address as usize,
+            IrqLine::alloc_specific(irq as u8).unwrap(),
+        );
+        enable_external_interrupt(irq as u16, 0xFE);
```

Comment:
I think it would be safer to use priority 1, if we don't need to prioritize them. The specification says "priority 1 is the lowest active priority while the maximum level of priority depends on PLIC implementation", so it's unclear that how PLIC implementation react to invalid priority.
```suggestion
        enable_external_interrupt(irq as u16, 1);
```